src/Microsoft.AspNet.Mvc.Core/ConsumesAttribute.cs

```diff
@@ -184,5 +189,15 @@ private List<MediaTypeHeaderValue> GetContentTypes(string firstArg, string[] arg
 
             return contentTypes;
         }
+
+        /// <inheritdoc />
+        public void SetContentTypes(IList<MediaTypeHeaderValue> contentTypes)
+        {
+            contentTypes.Clear();
```

Comment:
`[Consumes]` doesn't allow multiple attributes. `[Consumes("application/json", "application/xml")]` is how you would write it, and gives the desired behavior.